backend/hostgw/hostgw.go

```diff
@@ -145,8 +151,62 @@ func (rb *HostgwBackend) handleSubnetEvents(batch subnet.EventBatch) {
 				continue
 			}
 
+		case subnet.SubnetListed:
+			log.Info("Subnet listed: ", evt.Lease.Network)
+
+			if evt.Lease.Attrs.BackendType != "host-gw" {
+				log.Warningf("Ignoring non-host-gw subnet: type=%v", evt.Lease.Attrs.BackendType)
+				continue
+			}
+
+			subExistInRoute := checkSubnetExistInRouts(evt.Lease)
+
+			if !subExistInRoute {
+				route := netlink.Route{
+					Dst:       evt.Lease.Network.ToIPNet(),
+					Gw:        evt.Lease.Attrs.PublicIP.ToIP(),
+					LinkIndex: rb.extIface.Index,
+				}
+				if err := netlink.RouteAdd(&route); err != nil {
+					log.Errorf("Error adding route to %v via %v: %v", evt.Lease.Network, evt.Lease.Attrs.PublicIP, err)
+					continue
+				}
+			}
+
 		default:
 			log.Error("Internal error: unknown event type: ", int(evt.Type))
 		}
 	}
 }
+
+func checkSubnetExistInRouts(lease subnet.SubnetLease) bool {
+	routeList, err := netlink.RouteList(nil, netlink.FAMILY_V4)
```

Comment:
This ends up getting called multiple times every second (as many as there are subnets). It should be called once outside the loop.